rna_design/rna_synthesizer_n.py

```diff
@@ -0,0 +1,316 @@
+                                            pseudoknots=pseudoknots,
+                                            usedProgram=usedProgram,
+                                            pkprogram=pkprogram,
+                                            pkparameter=pkparameter,
+                                            HotKnots_PATH=HotKnots_PATH,
+                                            strategy=strategy,
+                                            noOfColonies=noOfColonies,
+                                            output_file=output_file,
+                                            py=py,
+                                            name=name,
+                                            verbose=verbose,
+                                            output_verbose=output_verbose,
+                                            seed=seed,
+                                            improve_procedure=improve_procedure,
+                                            Resets=Resets,
+                                            ants_per_selection=ants_per_selection,
+                                            ConvergenceCount=ConvergenceCount,
+                                            antsTerConv=antsTerConv,
+                                            alpha=alpha,
+                                            beta=beta,
+                                            ER=ER,
+                                            Cstrweight=Cstrweight,
+                                            Cgcweight=Cgcweight,
+                                            Cseqweight=Cseqweight,
+                                            omega=omega,
+                                            time=time)
+
+        self.pre_processor = PreProcessor(shape_type=shape_type,
+                                          energy_range=energy_range,
+                                          max_num=max_num,
+                                          split_components=split_components)
+
+        self.estimator = SGDClassifier()
+        self.vectorizer = Vectorizer()
+        self.synthesizer = RNASynth(estimator=self.estimator,
+                                    vectorizer=self.vectorizer,
+                                    designer=self.designer,
+                                    pre_processor=self.pre_processor,
+                                    constraint_extractor=self.constraint_extractor,
+                                    n_synthesized_sequences_per_seed_sequence=n_synthesized_sequences_per_seed_sequence,
+                                    instance_score_threshold=instance_score_threshold,
+                                    shuffle_order=shuffle_order,
+                                    negative_shuffle_ratio=negative_shuffle_ratio,
+                                    n_jobs=n_jobs,
+                                    cv=cv,
+                                    n_iter_search=n_iter_search)
+        logger.info('Created a RNASynthesizer object.')
```

Comment:
the logger can here get  a serialized version of the RNASynthesizer object
somthing like
logger.debug(str(self.synthesizer))
of course this means that the methods __repr__() are defined for RNASynthesizer and for each class contained in RNASynthesizer